### PR TITLE
Displaying Information From Spotify Player to User

### LINKED ISSUE #6: Add data about current song to player
Once we have web playback complete, we may want to know what song is currently playing in our UI.
This data can be fetched from the WebPlayback SDK getCurrentState method
https://developer.spotify.com/documentation/web-playback-sdk/reference/

This issue will be complete when our application displays the current song title, album title, and artist. If there is no current song, display a message saying no song is currently playing

### PR DESCRIPTION
**Pull request by 3Ustin**
Created a button for testing information from the player to the user. Learned about how unmounted hooks create errors and used that info to solve the player's state not being able to be updated. To solve the issue the team created an updatePlayerInfo function.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import logo from './logo.svg';
 import './App.css';
 import WebPlayback from './components/WebPlayBack';
+import PlayerUI from './components/PlayerUI'
```
Comment (MtuMartian) [inline src/App.tsx]: minor: This can be removed since we're not using the PlayerUI component in App.tsx

File: src/App.tsx [hunk 2]
```diff
@@ -12,6 +13,7 @@ function App() {
           Edit <code>src/App.tsx</code> and save to reload.
         </p>
         <WebPlayback/>
+
```
Comment (MtuMartian) [inline src/App.tsx]: minor: unnecessary white space

File: src/components/PlayerUI.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect} from 'react';
+import WebPlayback from './WebPlayBack';
+
+//Acceptable format for props data.
+export interface IPlayerUIProps {
+    trackName: string,
+    albumName: string,
+    artistName: string
+}
+//@ts-ignore
```
Comment (MtuMartian) [inline src/components/PlayerUI.tsx]: Do we still need this @ts-ignore?

File: src/components/PlayerUI.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const {
+        trackName,
+        albumName,
+        artistName
+    } = props;
+
+    //Displaying to the user props.
+    return(
+        <div> 
+            <div>Name: {trackName}</div>
```
Comment (MtuMartian) [inline src/components/PlayerUI.tsx:20]: I like this component! It's very simple and can be reused easily to display information

File: src/components/WebPlayBack.tsx [hunk 5]
```diff
@@ -15,32 +21,68 @@ function WebPlayback(props: any) {
         //'ts-ignore' is a Bandaid fix see issue on GitHub to create a 'Type Declaration File'.
         // @ts-ignore
         window.onSpotifyWebPlaybackSDKReady = () => {
-            const token = 'My Access Token';
-            //@ts-ignore
+            const token = 'BQCQQz1SLEaC7WIFgytE1srWTIIZSqZXgRPqWYE_CjH3Tgf4UAk-ennMkN8zAKaeQ22GLljT3F_T_EUjHUD2hqzIS8O8q92AwWIDsNa-5rMdyF3-MqcRnaJr377QW8E9k4GusUcFQk6glcDR4Kx9mq5aTPF1UBKM5QM';
```
Comment (MtuMartian) [inline src/components/WebPlayBack.tsx]: 🤔

File: src/components/WebPlayBack.tsx [hunk 6]
```diff
@@ -1,9 +1,15 @@
-import { useEffect } from 'react';
+import { useEffect, useState} from 'react';
+import PlayerUI, {IPlayerUIProps} from './PlayerUI';
 
 //In current build put Access Token Manually in (Line 18)
 //Look for an issue in Git for authentification for more info.
 function WebPlayback(props: any) {
+    //two state variables for updating Player Object
+    const [statePlayer, setPlayer] = useState();
```
Comment (MtuMartian) [inline src/components/WebPlayBack.tsx]: minor but I don't like the name statePlayer. Can we make it just player and change the player variable below to be spotifyPlayer or something like that?

File: src/components/WebPlayBack.tsx [hunk 7]
```diff
@@ -15,32 +21,68 @@ function WebPlayback(props: any) {
...
                 console.log('Device ID has gone offline', device_id);
             });
 
+            //@ts-ignore
+            player.addListener('player_state_changed', (state) => {
+                console.log(state);
+                const {track_window: { current_track }} = state;
+
+                //Creating playerInfo object for props
+                const updatedPlayerInfo: IPlayerUIProps = {
```
Comment (MtuMartian) [inline src/components/WebPlayBack.tsx]: We have some repeated logic here and on line 36 where we transform a state update to the player information that we need. Can we create one function for doing these and call it from `player.addListener('player_state_changed')` and `player.getCurrentState().then`?

File: src/components/WebPlayBack.tsx [hunk 8]
```diff
@@ -15,32 +21,68 @@ function WebPlayback(props: any) {
...
+            setPlayer(player);
         };
     }, []);
 
+    const updatePlayerInfo = (playerInfo: IPlayerUIProps) => {
+        setPlayerInfo(playerInfo);
+    }
+
     return (
+        /*THIS IS TEST CODE FOR RECIEVING PLAYER INFO THROUGH A BUTTON*/
```
Comment (MtuMartian) [inline src/components/WebPlayBack.tsx]: This comment should be removed

**Review by MtuMartian — COMMENTED** [review]
Overall looks good!

I left a few minor comments as well as a comment about repeated code in the WebPlayback.tsx component that should be addressed before merging

File: src/App.tsx (hunk 1 above)
Comment (3Ustin) [reply src/App.tsx]: Removed Import - was there for a PlayerUI Test and since test is gone no longer needed.

File: src/App.tsx (hunk 2 above)
Comment (3Ustin) [reply src/App.tsx]: gone

File: src/components/PlayerUI.tsx (hunk 3 above)
Comment (3Ustin) [reply src/components/PlayerUI.tsx]: Unnecessary, was there when IPlayerUIProps didn't exist - instead there was a Variables being passed through props as any. Removed.

File: src/components/PlayerUI.tsx (hunk 4 above)
Comment (3Ustin) [reply src/components/PlayerUI.tsx:20]: Thanks! Could you give an example? I might not have taken my medicine so I can't see it. How?

File: src/components/WebPlayBack.tsx (hunk 5 above)
Comment (3Ustin) [reply src/components/WebPlayBack.tsx]: 🤔

File: src/components/WebPlayBack.tsx (hunk 6 above)
Comment (3Ustin) [reply src/components/WebPlayBack.tsx]: I am slightly confused, though I agree this is a minor issue. Shouldn't we make it easy for future developers to know that this is related to the state? Is that like implicit? I guess I'm answering my own question I'll make this change. I actually just realized that it is in-fact implicit. Thanks!

Comment (3Ustin) [reply src/components/WebPlayBack.tsx]: Actually, I removed const [statePlayer,setPlayer] = useState(); AND setPlayer(player) Becuase we don't need the state information to display the player.

File: src/components/PlayerUI.tsx (hunk 4 above)
Comment (MtuMartian) [reply src/components/PlayerUI.tsx:20]: Oh I just liked that it separated the logic of getting the player information from displaying it :)

File: src/components/WebPlayBack.tsx (hunk 7 above)
Comment (3Ustin) [reply src/components/WebPlayBack.tsx]: Great! Thanks. I have now consolidated the code. Look at line 52 for a new update PlayerInfo method.

File: src/components/WebPlayBack.tsx (hunk 8 above)
Comment (3Ustin) [reply src/components/WebPlayBack.tsx]: Well said - this has been removed.

Comment (MtuMartian) [reply src/components/WebPlayBack.tsx]: Nice! :rocket:

File: src/components/WebPlayBack.tsx (hunk 6 above)
Comment (MtuMartian) [reply src/components/WebPlayBack.tsx]: Oh good call!

File: src/components/WebPlayBack.tsx [hunk 9]
```diff
@@ -53,30 +41,28 @@ function WebPlayback(props: any) {
...
             player.connect();
-
-            setPlayer(player);
         };
     }, []);
 
-    const updatePlayerInfo = (playerInfo: IPlayerUIProps) => {
-        setPlayerInfo(playerInfo);
+//@ts-ignore
+    const updatePlayerInfo = (state) => {
```
Comment (MtuMartian) [inline src/components/WebPlayBack.tsx:52]: You should be able to remove the ts-ignore here by saying `state: any`

File: src/components/WebPlayBack.tsx (hunk 7 above)
Comment (MtuMartian) [reply src/components/WebPlayBack.tsx]: Thanks for making the change! Looks good :smile:

**Review by MtuMartian — APPROVED** [review]
Looks great! I left one more minor comment about a ts-ignore that we can probably remove but feel free to squash + merge whenever!

**Pull request merged**